RHDEVDOCS-6022: Content creation for namespace-scoped Argo Rollouts installation (#78933)
openshift/openshift-docs: argo_rollouts/enable-support-for-namespace-scoped-argo-rollouts-installation.adoc

:_mod-docs-content-type: ASSEMBLY
include::_attributes/common-attributes.adoc[]
[id="enable-support-for-namespace-scoped-argo-rollouts-installation"]

> = Enable support for namespace-scoped Argo Rollouts installation

Reviewer: ```suggestion
= Enabling support for a namespace-scoped Argo Rollouts installation
```

Author: Fixed.

:context: enable-support-for-namespace-scoped-argo-rollouts-installation

toc::[]

> The {gitops-title} enables support for two modes of Argo Rollouts installation:

Reviewer: ```suggestion
{gitops-title} enables support for two modes of Argo Rollouts installations:
```
The names of products do not take articles (i.e., "OpenShift Container Platform", not "the OpenShift Container Platform".

Author: Fixed.

> * *Cluster-scoped installation* (default): The Argo Rollouts custom resources (CRs) defined in any namespace are reconciled by the Argo Rollouts instance. This allows you to use Argo Rollouts CR across any namespace on the cluster.

Reviewer: ```suggestion
* *Cluster-scoped installation* (default): The Argo Rollouts custom resources (CRs) defined in any namespace are reconciled by the Argo Rollouts instance. As a result, you can use Argo Rollouts CR across any namespace on the cluster.
```
Per IBM Style, avoid implying that software can "allow" humans to do anything. See https://www.ibm.com/docs/en/ibm-style?topic=grammar-anthropomorphism.

Author: Fixed.

> * *Namespace-scoped installation*: The Argo Rollouts instance is installed in a specific namespace and only handles Argo Rollouts CR within the same namespace. This installation mode includes the following benefits:

Reviewer: ```suggestion
* *Namespace-scoped installation*: The Argo Rollouts instance is installed in a specific namespace and only handles an Argo Rollouts CR within the same namespace. This installation mode includes the following benefits:
```

Author: Fixed.

> ** This mode does not require cluster-wide `ClusterRole` or `ClusterRoleBinding` permissions. Users may install and use Argo Rollouts within a single namespace without requiring cluster permissions.

Reviewer: ```suggestion
** This mode does not require cluster-wide `ClusterRole` or `ClusterRoleBinding` permissions. You can install and use Argo Rollouts within a single namespace without requiring cluster permissions.
```

Author: Fixed.

> ** This mode does not require cluster-wide `ClusterRole` or `ClusterRoleBinding` permissions. This means that only users with permissions that are limited to a specific namespace can use these resources.

Reviewer 2: 'can use these resources' -> 'can use Argo Rollouts resources'. 

At present, 'these' in the sentence seems like it could be referring to CR/CRB.

Or, if you prefer, another way of doing it:
'This mode does not require cluster-wide `ClusterRole` or `ClusterRoleBinding` permissions. Users may install and use Argo Rollouts within a single namespace without requiring these cluster permissions.'

> ** This mode does not require cluster-wide `cluster roles` or `cluster role binding` permissions. This means that only users with permissions that are limited to a specific namespace can use these resources.

Reviewer 3: ```suggestion
** This mode does not require cluster-wide `ClusterRole` or `ClusterRoleBinding` permissions. This means that only users with permissions that are limited to a specific namespace can use these resources.
```

Author: Fixed.

** This mode provides security benefits by limiting the cluster scope of a single Argo Rollouts instance to a specific namespace.

[NOTE]
====

> {gitops-title} does not allow both modes of Argo Rollouts installation at the same time to prevent unintended privilege escalation.

Reviewer 4: Reads a little like neither mode is allowed to prevent unintended privilege escalation. Maybe something like: "To prevent unintended privilege escalation, {gitops-title} allows only one mode of Argo Rollout installation at a time" instead.

Author: Fixed.

====

> Perform the following steps to switch between cluster-scoped and namespace-scoped Argo Rollouts installations.

Reviewer: ```suggestion
To switch between cluster-scoped and namespace-scoped Argo Rollouts installations, complete the following steps:
```

Reviewer: IBM Style Word Usage advises to avoid the term _perform_ in technical content where possible.

Author: Fixed.

:_mod-docs-content-type: ASSEMBLY
include::_attributes/common-attributes.adoc[]
[id="enable-support-for-namespace-scoped-argo-rollouts-installation"]
= Enable support for namespace-scoped Argo Rollouts installation
:context: enable-support-for-namespace-scoped-argo-rollouts-installation

toc::[]

By default, the {gitops-title} installation enables support for the cluster-scoped Argo Rollouts installation. In this mode of installation, the Argo Rollouts custom resources (CRs) defined in any namespace are reconciled by that Argo Rollouts instance. This allows you to use Argo Rollouts CR across any namespace on the cluster.

> The namespace-scoped Argo Rollouts installation is an alternative to the cluster-scoped Argo Rollouts installation. This mode of installation does not require cluster-wide permissions by using `cluster roles` or `cluster role binding`. This means that only users with permissions that are limited to a specific namespace can use such resources.

Author: Yes, they are cluster roles and cluster role binding.

A namespaced-scoped instance provides security benefits by limiting the cluster scope of a single Argo Rollouts instance. With this mode of installation, Argo Rollouts only handles Argo Rollouts CR within the same namespace.

> For example, if Argo Rollouts is installed in the 'argo-rollouts-ns' namespace, the Argo Rollouts installation only handles Argo Rollouts CR in the 'argo-rollouts-ns' namespace. The Argo Rollouts CR installed in another namespace, for example, the 'my-application' namespace, are not used.

Reviewer 2: How about this, making it slightly more clear that this paragraph continues to refer to namespace-scoped instances (I want to make sure that people who are only skimming are not confused):

For example, if **namespace-scoped** Argo Rollouts is installed in the 'argo-rollouts-ns' namespace, **that** Argo Rollouts installation only handles Argo Rollouts CR in the 'argo-rollouts-ns' namespace. **An** Argo Rollouts CR installed in another namespace, for example, the 'my-application' namespace, **is** not **handled**.

(changes are bolded)

Author: This feedback has been reworded as per the reviewer's comment. Fixed.

> For example, if Argo Rollouts is installed in the 'argo-rollouts-ns' namespace, the Argo Rollouts installation only handles Argo Rollouts CR in the 'argo-rollouts-ns' namespace. The Argo Rollouts CR installed in another namespace, for example, the 'my-application' namespace, are not used.

Author: Fixed.

> To configure namespace-scoped or cluster-scoped Argo Rollouts installation, you must use the `NAMESPACE_SCOPED_ARGO_ROLLOUTS` environment variable, which is set under the `.spec.config.env` property of the {gitops-title} Subscription CR. When this environment variable is set to `true`, Argo Rollouts provides support for namespace-scoped Argo Rollouts installation and when it is set to `false`, Argo Rollouts supports cluster-scoped  installation.

Reviewer 2: I would add this final sentence to this paragraph. It's important to be specific about the default:

If the value is not set, it defaults to `false` (cluster-scoped).

Author: This comment has been fixed as per the reviewer's comment. Fixed.

> To configure namespace-scoped or cluster-scoped Argo Rollouts installation, you must use the `NAMESPACE_SCOPED_ARGO_ROLLOUTS` environment variable, which is set under the `.spec.config.env` property of the {gitops-title} Subscription CR. When this environment variable is set to `true`, Argo Rollouts provides support for namespace-scoped Argo Rollouts installation and when it is set to `false`, Argo Rollouts supports cluster-scoped  installation.

Author: Fixed.

[NOTE]
====

> {gitops-title} is designed to prevent unintended privilege escalation by supporting the installation of either the cluster-scoped or the namespaced-scoped Argo Rollouts installation on a cluster. It does not allow both modes of Argo Rollouts installation, so you must choose which one to enable on the cluster, with the default being cluster-scoped.

Author: Fixed.

====

> The following steps are used to switch between a cluster-scoped installation to a namespace-scoped Argo Rollouts installation. Enabling support for namespace-scoped Argo Rollouts installation automatically disables the cluster-scoped installation.

Author: Fixed.
